index.js

```diff
@@ -132,33 +140,44 @@ function serialize(name, val, options) {
     var maxAge = opt.maxAge - 0;
 
     if (isNaN(maxAge) || !isFinite(maxAge)) {
-      throw new TypeError('option maxAge is invalid')
+      var maxAgeError = new TypeError('option maxAge is invalid')
+      maxAgeError.code = 'ERR_INVALID_ARG_VALUE'
+      throw maxAgeError;
     }
 
     str += '; Max-Age=' + Math.floor(maxAge);
   }
 
   if (opt.domain) {
     if (!fieldContentRegExp.test(opt.domain)) {
-      throw new TypeError('option domain is invalid');
+      var domainError = new TypeError('option domain is invalid')
+      domainError.code = 'ERR_INVALID_ARG_VALUE'
+      throw domainError;
     }
 
     str += '; Domain=' + opt.domain;
   }
 
   if (opt.path) {
     if (!fieldContentRegExp.test(opt.path)) {
-      throw new TypeError('option path is invalid');
+      var pathError = new TypeError('option path is invalid')
+      pathError.code = 'ERR_INVALID_ARG_VALUE'
+      throw pathError;
     }
 
     str += '; Path=' + opt.path;
   }
 
   if (opt.expires) {
     var expires = opt.expires
-
-    if (!isDate(expires) || isNaN(expires.valueOf())) {
-      throw new TypeError('option expires is invalid');
+    var expiresError = new TypeError('option expires is invalid')
```

Comment:
allocating an error object when it's not needed incurs a very small performance hit, due to the stack trace generation

This is one of those instances where following Dont Repeat Yourself isn't worth the perf cost. We save a tiny amount of repetition, but at a performance cost.